### PR TITLE
Enable generation of html report of integration tests, and upload to dokuwiki.

### PR DESCRIPTION
**Pull request by karandesai-96**
This PR introduces a set of pytest functions which would be complaint with [pytest-html](https://www.github.com/davehunt/pytest-html). These fixtures introduce a layer of customization over default html report provided by plugin.

* As mentioned in README of **pytest-html**, to generate a report, an extra argument has to be included as:
```
--html=path/to/file.html
```

* This command line option no longer has to be specified, due to invocation of `pytest_configure` method, before test execution. It assigns a _tempfile_  path to `config.option.filepath` for writing html report.

* Another method `pytest_unconfigure`, serves as a global teardown. It uploads the test report to dokuwiki.

* The command line argument `--slow-tests-data` has been renamed to `--integration-tests` and it accepts a path to YAML configuration for running slow tests. All other related fixtures have been modified. For now the schema of configuration file is:
```yml
reference: "/absolute/path/to/reference/dir"

dokuwiki:
  username: "username"
  password: "password"
```

This configuration file is parsed as a dict in a fixture created in **conftest.py**.

The checklist here maintains the status of work on this PR:
- [x] Show commit hash of the build on which current test run is being executed.
- [x] Make pytest-html write the html report to a tempfile, and `--html` may no longer be required.
- [x] Upload report to dokuwiki and leave no trace of tempfiles in filesystem.
- [x] Accept configuration for integration test run as YAML instad of individual fixtures.

### CODE REVIEW

**Comment by karandesai-96** [comment]
The resources referred were:
https://pytest.org/dev/plugins.html
https://pytest.org/dev/_modules/_pytest/hookspec.html
https://www.github.com/davehunt/pytest-html

Test execution looks like this:
image

Here is the report for fafa9e5, which is specified in the screenshot:
http://opensupernova.org/~karandesai96/integration2/doku.php?id=reports:fafa9e5

File: tardis/conftest.py [hunk 1]
```diff
@@ -35,6 +42,54 @@ def pytest_addoption(parser):
...
+def pytest_configure(config):
+    html_file = tempfile.NamedTemporaryFile(delete=False)
+    # Html test report will be generated at this filepath by pytest-html plugin
+    config.option.htmlpath = html_file.name
+
+
+def pytest_unconfigure(config):
+    # Html report created by pytest-html plugin is read here, uploaded to
+    # dokuwiki and finally deleted.
+    try:
```
Comment (wkerzendorf) [inline tardis/conftest.py]: you should do a try and except statement at the beginning with then just having dokuwiki_available boolean variable.

File: tardis/conftest.py [hunk 2]
```diff
@@ -35,6 +42,54 @@ def pytest_addoption(parser):
     parser.addoption("--slow-test-data", dest="slow-test-data",
```
Comment (wkerzendorf) [inline tardis/conftest.py]: I thought we changed this to integration test (at least @yeganer and I discussed this).

File: tardis/conftest.py [hunk 3]
```diff
@@ -35,6 +42,54 @@ def pytest_addoption(parser):
...
                      help="path to directory having baseline data for slow tests")
 
+    # Accept dokuwiki credentials through command line.
+    parser.addoption("--username", dest='username', help="Dokuwiki username")
```
Comment (wkerzendorf) [inline tardis/conftest.py]: I think you should make a yaml file that contains all of that information.

File: tardis/conftest.py [hunk 4]
```diff
@@ -35,6 +42,54 @@ def pytest_addoption(parser):
...
+        # Upload report on dokuwiki. Temporary link due to prototyping purposes.
+        doku_conn.pages.append("reports:{0}".format(githash[:7]), report_content)
+        print "Uploaded report of {0} on Dokuwiki".format(githash[:7])
+    except:
+        print "Dokuwiki python bindings not found!"
+
+    # Remove the local report file. Keeping the report saved on local filesystem
+    # is not desired, hence deleted.
+    os.unlink(config.option.htmlpath)
+    print "Deleted html report previously existing at {0}".format(
```
Comment (wkerzendorf) [inline tardis/conftest.py]: I think we can completely remove this or put this in the debug log.

File: tardis/conftest.py (hunk 1 above)
Comment (karandesai-96) [inline tardis/conftest.py]: I'll do it that way then.

File: tardis/conftest.py (hunk 4 above)
Comment (karandesai-96) [inline tardis/conftest.py]: I would remove `print` statements for now.

File: tardis/conftest.py (hunk 1 above)
Comment (wkerzendorf) [inline tardis/conftest.py]: the try-except block should only contain things that fail. Move this block up to the import statements and only use if here.

File: tardis/conftest.py [hunk 5]
```diff
@@ -32,8 +50,48 @@ def pytest_addoption(parser):
...
+
+            report_content = (
+                "Test executed on commit "
+                "[[https://www.github.com/tardis-sn/tardis/commit/{0}|{0}]]\n\n"
+                "{1}".format(githash, report_content)
+            )
+            # Upload report on dokuwiki. Temporary link due to prototyping purposes.
+            doku_conn.pages.append("reports:{0}".format(githash[:7]), report_content)
+        except gaierror, dokuwiki.DokuWikiError:
+            pass
```
Comment (wkerzendorf) [inline tardis/conftest.py]: again - stick the things that generate the error in the try except block and nothing else.

File: tardis/conftest.py [hunk 6]
```diff
@@ -1,5 +1,23 @@
...
+from tardis.atomic import AtomData
+from tardis.io.config_reader import Configuration
 from astropy.tests.pytest_plugins import *
 
+# For specifying error while exception handling
+from socket import gaierror
+
+try:
+    import dokuwiki
+    dokuwiki_available = True
```
Comment (wkerzendorf) [inline tardis/conftest.py]: same goes here - the things that raise the error need to be in the try, but nothing else.

File: tardis/conftest.py [hunk 7]
```diff
@@ -32,8 +51,50 @@ def pytest_addoption(parser):
...
+    if dokuwiki_available:
+        githash = tardis.__githash__
+        report_content = open(config.option.htmlpath, 'rb').read()
+        report_content = report_content.replace("<!DOCTYPE html>", "")
+
+        report_content = (
+            "Test executed on commit "
+            "[[https://www.github.com/tardis-sn/tardis/commit/{0}|{0}]]\n\n"
+            "{1}".format(githash, report_content)
+        )
```
Comment (karandesai-96) [inline tardis/conftest.py]: @wkerzendorf These four lines are good to be here or in `else` block ? I spoke to myself: _try to make a connection, except these two errors, else upload the report we have_. Hence, I kept the report ready beforehand (compromising the time of execution of these lines if connection isn't at all made.)

Comment (wkerzendorf) [inline tardis/conftest.py]: @karandesai-96 it's not about execution time it's about making the code easy to understand. What really happens when the if is True, what happens when the if is False. I think its fine the way it is for now.

File: tardis/conftest.py [hunk 8]
```diff
@@ -73,62 +134,60 @@ def pytest_report_header(config):
```
Comment (karandesai-96) [inline tardis/conftest.py]: @wkerzendorf Do we require a `try-except-else` for numpy as well ?

**Pull request merged**